### PR TITLE
Updates Facebook Graph api to latest + new fields

### PR DESCRIPTION
**Pull request by mduke**
- Adds permalink as an option to post node
- Adds children -> data as carouselImages to support carousels

### CODE REVIEW

**Comment by oorestisime** [comment]
Nice thanks for the contribution. Should we enable the carousel images to be used with gatsby image?
also this solves #132 ?

**Comment by mduke** [comment]
@oorestisime I will gladly add that functionality. Apologies for the delay I’m using your plugin for a personal project and the day job gets in the way of the fun stuff. I’ll make an update within the next 48 hours for you to review.

**Comment by mduke** [comment]
@oorestisime Since we last left off gatsby image is now supported within carousel images.

**Updates Included:**
1. Updates api endpoint to utilize 7.x
2. Adds permalink and carouselmage types
3. Rewrote normalization.js to support the creation of multiple local file nodes
4. Normalized naming conventions with what existed since the last commit (preview)
5. Updated README.md to show users how to use carousel images

**Comment by oorestisime** [comment]
Hey thanks for this!
I am currently on holiday for the next 2 weeks and won't be able to check
this!
Lookinh forward to merge this when i am back!

**Comment by mduke** [comment]
Have a great trip!

**Comment by oorestisime** [comment]
Hey sorry for the long wait.

Not sure why but lots of eslint errors. not sure if you used local eslint config or maybe your own? this provoked a lot of unwanted changes i think. Let me know if you need some help setting this up :)

File: src/instagram.js [hunk 1]
```diff
@@ -1,110 +1,101 @@
+"use strict";
+
+exports.__esModule = true;
```
Comment (oorestisime) [inline src/instagram.js]: not sure i understand this either (reference my other comment.). is it because of something automatic reformatting?

**Review by oorestisime — CHANGES_REQUESTED** [review]
Hey its pretty hard to review given all the reformatting that happened. i think it looks ok but could you please format as was initially with eslint and prettier config in the repo 🙇 ?

**Comment by oorestisime** [comment]
Also given you worked with the graph api i would like your input for https://github.com/oorestisime/gatsby-source-instagram#instagram-graph-api-token
Did you generate the token using this approach? was there anything not working?

File: src/instagram.js (hunk 1 above)
Comment (mduke) [reply src/instagram.js]: Hey, running prettier via the vscode plugin added some unexpected markup. I reformatted and then manually aligned what was a miss, all eslint errors have been corrected, corrected merge conflicts with the new hashtag work.

**Comment by mduke** [comment]
@oorestisime I used the instructions verbatim without issue

**Comment by mduke** [comment]
@oorestisime please let me know if you have any questions around my updates.

**Comment by oorestisime** [comment]
Hey yes i am planning to have a look at this tomorrow. Sorry for the wait :)

**Comment by oorestisime** [comment]
Hey sorry for the long delay. this looks great. i need to test and merge and release but i am not on my main pc so i need a bit more time! sorry about this. 
once again thanks for the PR

**Comment by mduke** [comment]
@oorestisime Any movement on getting this merged?

**Comment by oorestisime** [comment]
Hey once again sorry for the long delay., I returned yesterday to my main pc after being on my laptop for 2 weeks away from home. i ll be merging this in the next couple of days and  doing a release. 
once again sorry for the wait. really appreciate the contribution

**Comment by mduke** [comment]
@oorestisime I hope all is well. I see the latest changes have been merged (thank you) but the changes are not reflect in npm. Can you give this an npm publish so the latest can be seen here https://www.npmjs.com/package/gatsby-source-instagram.

**Comment by oorestisime** [comment]
Hey thanks for reminding me. just did

**Pull request merged**